src/lib/consentHandler/cmp.js

```diff
@@ -0,0 +1,66 @@
+export const GET_CONSENT_DATA = "getConsentData";
+export const GET_VENDOR_CONSENTS = "getVendorConsents";
+export const GET_SITE_CONSENT = "siteHasConsent";
+export const CMP_GET_MSG = '__cmpCall';
+export const CMP_RETURN_MSG = '__cmpReturn';
+export const CMP_FRAME = '__cmpLocator';
+export const CMP_API = '__cmp';
+
+
+/**
+ * Helper to determine GDPR consents.
+ *
+ * If consentConfig.type is 'iab', then it will attempt to call cmp in the following order:
+ *  1. Cmp in the same frame, or top frame.
+ *  2. If safe frame is used, then call cmp thru safe frame.
+ *  3. Look for a cmp frame and call cmp thru messages.
+ *
+ * If consentConfig.type is 'static', then the driver data is expected to be supplied in
+ * driver.consentData.  Usually this means the caller has managed to call cmp on its own.
+ */
+export default class Cmp {
+    constructor() {
+        this.returnMsgName = CMP_RETURN_MSG;
+        this.locatorFrame = CMP_FRAME;
+        this.cmpApi = CMP_API;
+        this.safeframeCall = "cmp";
+    }
+
+    getConsentCmd() {
+        return [[GET_CONSENT_DATA],[GET_SITE_CONSENT]];
+    }
+
+    getSafeFrameData(data) {
+        return (data.cmpCommand === GET_CONSENT_DATA) ?
+            data.vendorConsentData : data.vendorConsents;
+    }
+
+    setConsentData(value) {
+        const gdprApplies = (value[GET_CONSENT_DATA]) ? value[GET_CONSENT_DATA].gdprApplies : null;
+        const consentString = (value[GET_CONSENT_DATA]) ? value[GET_CONSENT_DATA].consentData : null;
+        const siteConsent = (value[GET_SITE_CONSENT]) ? value[GET_SITE_CONSENT] : null;
+        this.consentData = [{gdprApplies: gdprApplies, consentData: consentString},{siteConsent}];
+    }
+
+    getConsentData(result) {
```

Comment:
Because there are too many similarly named functions, maybe a more descriptive name would be _interpretConsentData_ or _normalizeConsentData_.

Reply:
cleaned up the naming while getting rid of the static handler